src/simulator.zig

```diff
@@ -632,30 +632,50 @@ pub const Simulator = struct {
         }
     }
 
-    // Check if the core is stuck in a view change because replicas can't converge to a canonical
-    // checkpoint. This can happen in two scenarios:
+    // Check if the core is stuck because replicas can't converge to a canonical checkpoint. This
+    // can happen in two scenarios:
     //
-    // - There is a canonical checkpoint, but replicas can't learn about because commit messages
-    //   don't flow during view change.
-    // - There is no canonical checkpoint at all --- one replica is ahead of others by a checkpoint,
-    //   but there are no commits on top of that checkpoint.
+    // A) There is no canonical checkpoint at all --- one replica is ahead of others by a
+    //    checkpoint, but there are no commits on top of that checkpoint.
+    // B) During view change, there is a canonical checkpoint, but replicas can't learn about
+    //    because commit messages don't flow during view change.
```

Comment:
Adjusted comment:

* there _aren't_ enough replicas in core at a given checkpoint, so they can't prove checkpoint canonical from pings
* and they also can't learn about canonicity from commit messages, because there aren't any